clusterloader2/pkg/flags/flags.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	noInt    = 0
+	noString = ""
+)
+
+func init() {
+	pflag.CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+}
+
+type flagEnv interface {
+	ParseEnv() error
+}
+
+type stringFlagEnv struct {
+	v            *string
+	envVariable  string
+	defaultValue string
+}
+
+// ParseEnv assigns env variable value if needed.
+func (s *stringFlagEnv) ParseEnv() error {
+	if *s.v != noString {
+		return nil
+	}
+	if s.envVariable != "" {
+		if val, ok := os.LookupEnv(s.envVariable); ok {
+			*s.v = val
+			return nil
+		}
+	}
+	*s.v = s.defaultValue
```

Comment:
Actually, I just realized that it doesn't work.
What if we say default is 5, but I set "--my-flag=0". It will then be overwritten to 5.

I think that this can be solved differently.
- you first parse EnvVars (with defaulting to default value)
- then you parse flags (to overwrite stuff)

That would solve your problem, right?

Reply:
The problem is we don't know whether the flag was provided or not - parse flag always returns value. If someone set "--my-flag=0" it will return the same value as default. In this case we don't know if we should override the env var.

Reply:
I saw that flag module provides interface `flag.Value` that handles setting. I need to verify is pflag also supports this interface.

Reply:
Did you see the flag.Var() as described here:
https://godoc.org/github.com/spf13/pflag

It seems that it's pretty much what you need.

Reply:
Ok. I changed flag declaration to flag.Var().
I also moved inner structures definitions to types.go.